apisix/admin/plugins.lua

```diff
@@ -42,7 +44,14 @@ end
 
 function _M.get(name)
     local arg = get_uri_args()
-    if arg and arg["all"] == "true" then
+    -- If subsystem is passed inside args then it should be oneOf: http / stream.
+    local subsystem = arg["subsystem"]
+    if subsystem and subsystem ~= "http" and subsystem ~= "stream" then
+        return 400, {error_msg = "unsupported subsystem: "..subsystem}
+    end
+
+    -- When name is unspecified, return all plugins for the given subsystem
```

Comment:
Why add this logic, it is not in the proposal

Reply:
This is present in the proposal document on 3.2.

Reply:
So the output of /apisix/admin/plugins?subsystem=xyz would be all plugins of subsystem xyz

Reply:
We have `/apisix/admin/plugins/list`, so we don't need this interface `/apisix/admin/plugins`

Reply:
`/apisix/admin/plugins` will return what `/apisix/admin/plugins?all=true` returns now. With full schema.

Reply:
The full schema is no needed any more